### PR TITLE
[Resolves #49] crud actions lifehack

### LINKED ISSUE #49: add complete CRUD actions for lifehacks
Create, Read, Update, Delete or Reviews & Lifehacks

### PR DESCRIPTION
**Pull request by katilaw**
#49
#88
#51

### CODE REVIEW

File: spec/features/lifehacks/04_user_updates_lifehack_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    expect(find_field('lifehack[description]').value).to eq(description)
+  end
+
+  scenario 'user attempts to edit another user\'s lifehack' do
+    user_sign_in(user2)
+    click_link(lh.title)
+    click_link('Edit Lifehack')
+
+    expect(page).to have_content('You are not the Authorized User')
+  end
```
Comment (cmkoller) [inline spec/features/lifehacks/04_user_updates_lifehack_spec.rb:65]: There should be another test here making sure that an admin user can edit another user's lifehack, if that's the desired functionality!

File: spec/features/lifehacks/05_user_deletes_lifehack_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    click_link(lifehack2.title)
+    click_button('Delete Lifehack')
+
+    expect(page).to have_content('You are not the Authorized User')
+    expect(current_path).to eq(root_path)
+    expect(page).to have_content(lifehack2.title)
+  end
+
+  scenario 'check to see if reviews are deleted with lifehack'
```
Comment (cmkoller) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb]: This scenario should either be filled out or not included at all!

File: spec/features/lifehacks/05_user_deletes_lifehack_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+end
```
Comment (cmkoller) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb:43]: There should be another test here checking to make sure an admin user can delete items!

Comment (katilaw) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb:43]: We test for this situation under the admin_spec

File: spec/features/lifehacks/05_user_deletes_lifehack_spec.rb (hunk 2 above)
Comment (katilaw) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb]: We usually leave some pending tests for future tests, this gets tested under the review tests. Which I'm currently writing. We just don't want to forget about such situations

Comment (katilaw) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb]: Actually, I'll write another test for that here.

File: spec/features/lifehacks/05_user_deletes_lifehack_spec.rb (hunk 3 above)
Comment (cmkoller) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb:43]: That's fine, my first two comments still stand though!

File: spec/features/lifehacks/05_user_deletes_lifehack_spec.rb (hunk 2 above)
Comment (cmkoller) [inline spec/features/lifehacks/05_user_deletes_lifehack_spec.rb]: It's better not to commit pending tests, just like it's better not to commit commented-out code. If you need to remember to do something, add it to whatever list you're using to keep track of your work elsewhere!

**Pull request merged**